### PR TITLE
Refactor layout for docs: issue #6391

### PR DESCRIPTION
**Pull request by inertial-frame**
Addresses: https://github.com/gravitational/teleport/issues/6391 - Restructuring of Docs in light of 6.2, new features, and the Edit Pass.

For simplicity's sake, the subsections are omitted in this branch until a permanent article-level structure is decided upon.

This PR adds several other alterations beyond Issue #6391. For example:

1. https://github.com/gravitational/teleport/pull/6929
2. Generally, I think we should strive for short articles but more of them rather than a few lengthy articles. We can be a little more aggressive about our tree TOC.

---

Builds off: https://github.com/gravitational/teleport/pull/7000 for https://github.com/gravitational/teleport/issues/6391 and https://github.com/gravitational/teleport/issues/6943.

### CODE REVIEW

**Comment by inertial-frame** [comment]
@stevenGravy - Adding you for visibility and from our discussion. Thanks for your suggestions about brevity of content - I've added that as a goal.

**Comment by inertial-frame** [comment]
Config article in Setup Section should include ACME notes. Unite all docs for that there.

File: docs/layout.md [hunk 1]
```diff
@@ -34,866 +34,148 @@ Along with a fifth for:
...
+- [x] Home (Renamed from documentation)
+    - [x] Introduction
+    - [x] Adopters
+    - [x] Getting Started
+    - [x] FAQ
+    - [x] Changelog
+- [ ] Setup (renamed from infrastructure guides)
+    - [x] Installation (moved from Home Section)
+    - [x] Docker Compose
+    - [ ] Cloud Infrastructure
```
Comment (inertial-frame) [inline docs/layout.md:46]: I'm not sure if we ruled decisively on where these should go ... I vote for here since these will be much briefer.

**Comment by russjones** [comment]
Let's close this for now, if we want to re-use it, let's create another PR against `master`.

**Pull request closed without merging**